### ISSUE TITLE
Passing options for a particular column to the client side?

### ISSUE THREAD

**Issue by doganmeh** [pivotal-energy-solutions/django-datatable-view#140]
Is it possible to add/override column specific options that go to the client side? Such as column [width ](https://datatables.net/reference/option/columns.width)or [render](https://datatables.net/reference/option/columns.render)?

If not, I think this would be a good candidate for the next major release.

**Comment by tiliv** [comment]
There are a couple of options, but right now are as direct as a kwarg to the Column constructor.  I need to allow ``attributes`` so that you have some power of it.  Without that, there isn't a clean way to just subclass the ``Column`` and allow the keyword, since all of the built-in columns we provide wouldn't benefit from it.

The ways to handle this is right now are to leverage css rules targeting ``th[data-name="column-name-slug"]``, or to use the table's javascript initialization step to send arbitrary other datatables.js options.

Although specifying client-side JS actions on the Django side of things is awkward, it's still on my radar to find a solution that doesn't require the developer to manage html, css, and python just for this kind of thing.

``render`` is not familiar to me, given that I only nominally brought this package up to speed for datatables.js 1.10, but I will look at it to see if there's something this package can do to offer better support than sending you to the js initialization hook.

**Comment by doganmeh** [comment]
> since all of the built-in columns we provide wouldn't benefit from it.

Why not? being able to set column width might be needed once in a while for any column, I think..

> or to use the table's javascript initialization step to send arbitrary other datatables.js options

Do you mean, something along these lines:

    datatableview.initialize($('.datatable'),  {
      "columnDefs": [
        { "width": "20%", "targets": 0 }
        { "width": "30%", "targets": 0 }
        { "width": "10%", "targets": 0 }
      ]
    } );

Would this work?

**Comment by doganmeh** [comment]
If that works, in a similar fashion, I should be able to do this as well:

    datatableview.initialize($('.datatable'),  {
      "columns": [
        { },
        { },
        { "render": my_function }
      ]
    } );

In that case, I just need to be able to guestimate which field goes where.

**Comment by tiliv** [comment]
>> since all of the built-in columns we provide wouldn't benefit from it.

> Why not? being able to set column width might be needed once in a while for any column, I think..

I agree, I just mean that subclassing Column doesn't help all of the existing subclasses of Column, so you would have to subclass all of those instead.  There's no technical reason why you can't do this, but it's far from ideal.

You are correct concerning the JS initialize hook.  Our ``initialize()`` method adds its own ``aaSorting`` and ``aoColumns`` (since this originated before the new api).  I think using ``columns`` will preempt ``aoColumns``, which might make some of our automatic behaviors vanish unless you reproduce them in ``columns``.  This will definitely have to be updated for the release.

**Comment by doganmeh** [comment]
I see.

**Comment by n6151h** [comment]
The `initialize()` method doesn't seem to apply the options object. :/

~~~javascript
<script type="text/javascript">
    $(function() {
        
        var dtOpts = {
                "lengthMenu": [[20], ["20"]],
                "colummns": [ 
                    {'width': '200px', "data": "date", "name": "date", "render": function(d, t, r, m) { console.log(d); return("xxx"); }},
                    {'width': '10%'},
                    {'width': '10%'},
                    {'width': '10%'},
                    {'width': '10%'},
                    {'width': '10%'},
                    {'width': '10%'},
                    {'width': '10%'},
                    {'width': '10%'}
                  ],
                "dom": ''
            };
            
        var dt = datatableview.initialize($(".dataTable"), dtOpts);
            
        console.log("store-daily-data initialized");
    });
</script>
~~~
